core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1741,21 +1744,21 @@ object Parser extends ParserInstances {
       }
     }
 
-    case class Fail[A]() extends Parser1[A] {
+    case class Fail[A]() extends Parser[A] {
       override def parseMut(state: State): A = {
         state.error = Chain.one(Expectation.Fail(state.offset));
         null.asInstanceOf[A]
       }
     }
 
-    case class FailWith[A](message: String) extends Parser1[A] {
+    case class FailWith[A](message: String) extends Parser[A] {
       override def parseMut(state: State): A = {
         state.error = Chain.one(Expectation.FailWith(state.offset, message));
         null.asInstanceOf[A]
       }
     }
 
-    final def oneOf[A](all: Array[Parser[A]], state: State): A = {
+    final def oneOf0[A](all: Array[Parser0[A]], state: State): A = {
```

Comment:
I don't think this name needs to change, there is only one internal implementation.